### ISSUE TITLE
uploading very large pcap files may fail

### ISSUE THREAD

**Issue by mmguero** [cisagov/Malcolm#101]
I've noticed that uploading very large pcap files can fail. I think this is a timeout somewhere in PHP on the upload container or in nginx itself.

**Comment by Jyasapara** [comment]
I have also noticed this problem when uploading a 10 GB file to Malcolm.

**Comment by mmguero** [comment]
As I work on this, just FYI for anyone hitting it: a workaround is to copy very large PCAP files directly into the `pcap/upload` directory, bypassing the upload interface completely, or using the SFTP upload.

I'm playing around with different buffer sizes and the like in the nginx config, I hope to have this figured out soon.

**Comment by mmguero** [comment]
I've been playing around with the settings I committed in a26691a (tweaked some things and disabled an intermediate buffer between the reverse proxy and the upload server) and I have been able to successfully upload various pcaps from 2 to 8 gigabytes without getting the error. I also merged seventeen 1-gigabyte PCAP files (Malcolm has a 20 GB single file upload limit in the configuration) with mergecap into one giant PCAP file and uploaded it. It took a while at the "100%" point as it assembles the file chunks, but it did eventually finish without an error, and the md5sums matched.

Marking as fixed for the next release (1.8.2 or 1.9.0, depending on what I get in there). Will reopen if I run into it again.
